src/software/amazon/ion/impl/LocalSymbolTableImportAdapter.java

```diff
@@ -89,101 +65,61 @@ public SymbolTable getSystemSymbolTable()
-        {
-            importedTables = imports.toArray(new SymbolTable[imports.size()]);
-        }
-
-        return importedTables;
+        return imports;
     }
 
     @Override
     public int getImportedMaxId()
     {
-        return maxImportedId;
+        return getDelegate().getImportedMaxId() - systemSymbolCount;
     }
 
     @Override
     public int getMaxId()
     {
-        return maxId;
-    }
-
-    @Override
-    public SymbolToken intern(String text)
-    {
-        throw new ReadOnlyValueException(this.getClass());
+        return getDelegate().getMaxId() - systemSymbolCount;
     }
 
     @Override
     public SymbolToken find(String text)
     {
-        for (SymbolTable importedTable : imports)
+        SymbolToken symbolToken = getDelegate().find(text);
+        if(symbolToken != null)
         {
-            SymbolToken symbolToken = importedTable.find(text);
-
-            if (symbolToken != null)
-            {
-                return symbolToken;
-            }
+            symbolToken = new SymbolTokenImpl(symbolToken.getText(), symbolToken.getSid() - systemSymbolCount);
         }
 
-        Integer sid = getLocalTextToSidMap().get(text);
-        if (sid != null)
-        {
-            return new SymbolTokenImpl(text, sid);
-        }
-
-        return null;
+        return symbolToken;
     }
 
     @Override
     public int findSymbol(String name)
     {
-        SymbolToken token = find(name);
-        return (token == null ? UNKNOWN_SYMBOL_ID : token.getSid());
+        int sid = getDelegate().findSymbol(name);
+        if(sid != UNKNOWN_SYMBOL_ID){
+            sid = sid - systemSymbolCount;
+        }
+
+        return sid;
     }
 
     @Override
     public String findKnownSymbol(int id)
     {
-        if (id < 0)
-        {
-            throw new IllegalArgumentException("symbol IDs must be greater than 0");
-        }
-
-        if (id > getMaxId()) // not in this table or imports
-        {
-            return null;
-        }
-
-        // look into imports
-        if (id <= maxImportedId)
-        {
-            for (SymbolTable importedSymbolTable : imports)
-            {
-                String text = importedSymbolTable.findKnownSymbol(id);
-                if (text != null)
-                {
-                    return text;
-                }
-            }
-        }
-
-        // not in imports get from local
-        int index = id - maxImportedId - 1;
-        return getSidToLocalText()[index];
+        return getDelegate().findKnownSymbol(id + systemSymbolCount);
     }
 
-    private List<SymbolTable> createImportList(LocalSymbolTable delegate)
+    private SymbolTable[] createImports(LocalSymbolTable delegate)
     {
-        SymbolTable[] asArray = delegate.getImportedTables();
-        List<SymbolTable> imports = new ArrayList<SymbolTable>(asArray.length);
-        for (SymbolTable s : asArray)
+        SymbolTable[] delegateImports = delegate.getImportedTables();
```

Comment:
The system symbol table is guaranteed to be the delegate's first import, right? Would asserting that and then just keeping the rest of the imports starting at index 1 allow you to simplify this?

Also, since this is only used as the return from `getImportedTables`, would it make sense to do this lazily?